Source/carthage/Build.swift

```diff
@@ -163,34 +163,43 @@ public struct BuildCommand: CommandProtocol {
 		}
 	}
 
-	/// Opens a temporary file on disk, returning a handle and the URL to the
+	/// Opens an existing file, if provided, or creates a temporary file if not, returning a handle and the URL to the
 	/// file.
-	private func openLogFile(_ path: String?) -> SignalProducer<(FileHandle, URL), NSError> {
+	private func openLogFile(_ path: String?) -> SignalProducer<(FileHandle, URL), CarthageError> {
 		return SignalProducer.attempt {
-			let logFD: Int32
-			let logPath: String
-			
 			if let path = path {
-				logFD = FileHandle(forUpdatingAtPath: path)?.fileDescriptor ?? -1
-				logPath = path
+				if !FileManager.default.fileExists(atPath: path) {
+					FileManager.default.createFile(atPath: path, contents: nil, attributes: nil)
+				}
+				
+				let fileURL = URL(fileURLWithPath: path, isDirectory: false)
+				
+				guard let handle = FileHandle(forUpdatingAtPath: path) else {
```

Comment:
@mdiep the `fileExists` check guards not-so-much the handle, but more the `createFile(atPath:…)` call, which [can overwrite: “If a file already exists at path, this method overwrites the contents of that file if the current process has the appropriate privileges to do so.”](https://developer.apple.com/reference/foundation/filemanager/1410695-createfile)

[`FileHandle.init(forUpdatingAtPath:)`](https://developer.apple.com/reference/foundation/filehandle/1411131-init) (from the description at least) is a superset of the behavior of [`FileHandle.init(forWritingAtPath:)`](https://developer.apple.com/reference/foundation/filehandle/1414405-init).